pkg/kubectl/cmd/util/factory.go

```diff
@@ -301,9 +323,36 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return priorityRESTMapper, api.Scheme
 		},
+		DynamicObject: func() (meta.RESTMapper, runtime.ObjectTyper) {
```

Comment:
This should probably return an error (that a command has to handle) - nothing else in factory calls CheckErr().  Does it complicate where this will be called from?

I do think we want to make both of these interfaces somewhat lazy (only load config when asked, and cache).  If we do it this way, will we always be doing server discovery in every get call?  Does that make normal get significantly worse in terms of number of requests it is making to the server?